addons/html_builder/static/tests/custom_tab/builder_components/builder_many2many.test.js

```diff
@@ -68,3 +73,47 @@ test("many2many: find tag, select tag, unselect tag", async () => {
     expect("table tr").toHaveCount(1);
     expect("table input").toHaveValue("Second");
 });
+
+test("many2many: async load", async () => {
+    onRpc("/web/dataset/call_kw/test/name_search", async (args) => [
+        [1, "First"],
+        [2, "Second"],
+        [3, "Third"],
+    ]);
+    addActionOption({
+        testAction: {
+            load: ({ value }) => {
+                expect.step("load");
+                return new Promise((resolve) => setTimeout(() => resolve(value), 50));
```

Comment:
If you to simulate async stuff, you should use new Deferred(). it give you a way to control the timing.